File: 78. Subsets/main.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> sub;
        int n=nums.size();
        for(int i=0;i<pow(2,n);i++){
            int k=i;
            vector<int> p;
            for(int j=0;j<n;j++){
                if(k & 1)
                    p.push_back(nums[j]);
                k>>=1;
            }
            sub.push_back(p);
        }
            
        return sub;
        
    }
};

File: 78. Subsets/main.py
class Solution:
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        n=len(nums)
        sub=[]
        for i in range(2**n):
            #l=[0 for k in range(n)]
            k=i
            p=[]
            for j in range(n):
                if (k & 1):
                    p.append(nums[j])
                k>>=1
                
            sub.append(p)
            
        return sub
